packages/components/src/Menu/Menu.tsx

```diff
@@ -275,6 +336,104 @@ export function Menu({
   }
 }
 
+interface MenuComposableProps {
+  readonly children: ReactNode;
+  readonly UNSAFE_className?: {
+    menu?: string;
+  };
+  readonly UNSAFE_style?: {
+    menu?: CSSProperties;
+  };
+}
+
+function MenuComposable({ children }: MenuComposableProps) {
+  // Use positional arguments to determine the trigger and content
+  // Avoids parsing/iterating over the children
+  const [trigger, menu] = React.Children.toArray(children);
+
+  type AnimationState = "unmounted" | "hidden" | "visible";
+  const [animation, setAnimation] = useState<AnimationState>("unmounted");
```

Comment:
this is much more complicated than I'd like. there should be a way to make this much cleaner, however we need to untangle some `framer-motion` vs `motion` business and likely update our version?

https://react-spectrum.adobe.com/react-aria/styling.html#framer-motion

https://motion.dev/blog/framer-motion-is-now-independent-introducing-motion

`framer-motion` is now just `motion`